feat(marketplace): review of marketplace contribution rules and introduction to versioning (#1404)
mia-platform/documentation: docs/marketplace/add_to_marketplace/create_your_company_marketplace.md

Refer to the [related miactl documentation](/cli/miactl/30_commands.md#apply) to know the exact specifications of such object.
:::

### Set controlled versions for your Marketplace resources and make them available to users

:::info
This feature is currently available for the [Plugin](/marketplace/add_to_marketplace/add_item_by_type/add_plugin.md) and [Custom Resource](/marketplace/add_to_marketplace/add_item_by_type/add_custom_resource.md) types.
:::

Marketplace creators have now the possibility to manage some types of resources (at the moment, Plugins and Custom Resources of type k8s) through a governance based on a versioning system.

This means that, for these types of resources, it is possible to establish a more structured and transparent lifecycle management system, providing users access to all the versions of a resource and allowing them to see the release notes of each version and select and instantiate the version that best suits their configuration needs among the ones made available by the Marketplace creator.

> This versioning system also provides significant advantages in a feature-branch design workflow, effectively leveragin different versions of the same marketplace item across various branches of a project.

Author: ```suggestion
This versioning system also provides significant advantages in a feature-branch design workflow, effectively leveraging different versions of the same marketplace item across various branches of a project.
```

> For instance, we imagine to have a project with three different branches: *dev*, *stage* and *main*, with the latter being the branch with the official configuration. Let's also imagine to have installed in all the three services the *CRUD Service* plugin, which includes different versions.

Author: ```suggestion
For instance, we imagine to have a project with three different branches: *dev*, *stage* and *main*, with the latter being the branch with the official configuration. Let's also imagine to have installed in all the three branches the *CRUD Service* plugin, which includes different versions.
```

Reviewer: On a second thought, better use `revisions`...

```suggestion
For instance, we imagine to have a project with three different revisions: *dev*, *stage* and *main*, with the latter being the branch with the official configuration. Let's also imagine to have installed in all the three revisions the *CRUD Service* plugin, which includes different versions.
```
